Don't require password confirmation if the password hasn't changed

ferblape opened the pull request:
Connects to #305

### What does this PR do?

This PR fixes admin update error that required to introduce a new password each time.

Commit:
Don't require password confirmation if the password hasn't changed

danguita commented on app/forms/gobierto_admin/admin_form.rb:
```diff
@@ -23,7 +23,8 @@ class AdminForm
     validates :email, format: { with: Admin::EMAIL_ADDRESS_REGEXP }
-    validates :password, presence: true, confirmation: true
+    validates :password, presence: true, if: Proc.new { |f| f.admin.new_record? }
+    validates :password, confirmation: true, if: Proc.new { |f|  f.password != f.admin.password }
```
Hey, sorry for intruding here, but wanted to point out that the conditional password confirmation check may not be needed here. I mean that the default `confirmation` validation behavior is taking the field presence into account. Also, I'd say we can take advantage of the `new_record?` property which is already there to condition the password presence check.

Let me please add a couple of commits to discuss that further.

danguita replied:
@ferblape because code is worth than thousand words, I've just pushed my thoughts here 😛 Would you say this is still readable and should work as expected?

danguita committed:
Slightly change the Admin password validation logic

- Make use of the `new_record?` property instead of adding a new check.
- Initialize `new_record?` property before doing any validations.
- Increase test coverage.

danguita committed:
Cover missing and not matching passwords scenarios in Admin update tests

danguita committed:
Remove unused constant ¯\_(ツ)_/¯

ferblape commented:
Hey @danguita great to see you around! Yes, your approach is nicer and makes sense to use `new_record?` here. Thanks!
